chia/wallet/wallet_state_manager.py

```diff
@@ -876,12 +876,16 @@ async def new_coin_state(
         used_up_to = -1
         ph_to_index_cache: LRUCache = LRUCache(100)
 
-        for coin_state in coin_states:
+        local_records: List[Optional[WalletCoinRecord]] = await self.coin_store.get_coin_records(
+            [st.coin.name() for st in coin_states]
```

Comment:
this makes us compute the coin name twice. It's still a net improvement though, and there's no simple way of avoiding it. The `WalletCoinRecord` does not contain a name field.

Reply:
you could build a dict lookup for names, but probably marginal